### PR TITLE
Support conditional values in Vega

### PR DESCRIPTION
**Pull request by lukaw3d**
Changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/vega/vega/wiki/Marks#production-rules

### CODE REVIEW

**Comment by pelotom** [comment]
Why not just add the `test` field directly to the `ValueRef` interface?

**Comment by lukaw3d** [comment]
Simplified by adding into ValueRef

File: types/vega/index.d.ts [hunk 1]
```diff
@@ -524,8 +524,11 @@ declare namespace vg {
     mult?: number;
     offset?: number;
     band?: boolean;
+    test?: string;
   }
 
+  export type ValueRefs = ValueRef | ValueRef[];
```
Comment (lukaw3d) [inline types/vega/index.d.ts]: I'm having trouble deciding a good name for the extra type

Comment (pelotom) [reply types/vega/index.d.ts]: `ValueRefs` seems fine to me. Is it really the case that all aesthetics can take an array of value refs, not just a single one? (I haven't used vega in a while).

Comment (lukaw3d) [reply types/vega/index.d.ts]: Seems so; tried: `x`, `y2`, `width`, `fill`, `fillOpacity`, `stroke`

Comment (lukaw3d) [reply types/vega/index.d.ts]: The first value in the array that has no test or that has a passing test - is used

**Comment by pelotom** [comment]
:+1:

**Pull request merged**